moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1589,9 +1608,8 @@ bool TrajectoryExecutionManager::waitForRobotToStop(const TrajectoryExecutionCon
 
     // check for motion in effected joints of execution context
     bool moved = false;
-    for (const auto& trajectory : context.trajectory_parts_)
+    (const auto& joint_names : joint_names_per_traj)
```

Comment:
If you already have access to `context.trajectory_parts_` here, maybe consider locking and computing the joint names per traj in this function?

This means you wouldn't have to change the function interface in this case.